### PR TITLE
dev: add supabase cli tools

### PR DESCRIPTION
**Pull request by glebchanskiy**

### CODE REVIEW

**Comment by glebchanskiy** [comment]
* **#16** <a href="https://app.graphite.dev/github/pr/variants-community/variants-world/16?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @glebchanskiy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/db/supabase/types.ts [hunk 1]
```diff
@@ -1,6 +1,31 @@
 export type Json = string | number | boolean | null | { [key: string]: Json | undefined } | Json[]
 
 export type Database = {
+  graphql_public: {
```
Comment (glebchanskiy) [inline src/db/supabase/types.ts:4]: so as not to interfere with future supabase:gen

Comment (ArturSharapov) [reply src/db/supabase/types.ts:4]: wdym? or is it just generated supabase types included into this commit?

File: supabase/migrations/20240612140832_data.sql
Comment (ArturSharapov) [inline supabase/migrations/20240612140832_data.sql:1]: Is it the entire backup? We better not have it as a part of git history.
Either include a minimal version (e.g. 20 posts with its users/comments), or do not include at all.

We should setup automated backups and pull data from the artifacts produced.

File: package.json [hunk 2]
```diff
@@ -3,20 +3,21 @@
...
-    "migrate": "dotenv -e .env.development -- prisma migrate dev --create-only",
-    "migrate:dev": "dotenv -e .env.development -- prisma migrate deploy",
+    "gen:prisma:dev": "rm -rf node_modules/.prisma; dotenv -e .env.local -- prisma generate",
+    "gen:prisma:prod": "rm -rf node_modules/.prisma; PRISMA_OUTPUT=./generated deno run -A --node-modules-dir=false npm:prisma@5.4.1 generate --no-engine; cd prisma/generated/deno; cp edge.ts index.js",
+    "gen:prisma": "rm -rf node_modules/.prisma; dotenv -e .env.local -- prisma generate --no-engine; PRISMA_OUTPUT=./generated deno run -A --node-modules-dir=false npm:prisma@5.4.1 generate --no-engine; cd prisma/generated/deno; cp edge.ts index.js",
+    "gen:supabase": "npx supabase gen types typescript --local > src/db/supabase/types.ts",
+    "postinstall": "pnpm postinstall:${VW_ENV:-dev}",
+    "postinstall:dev": "pnpm gen:prisma:dev",
+    "postinstall:prod": "pnpm gen:prisma:prod",
     "migrate:production": "dotenv -e .env.production -- prisma migrate deploy",
```
Comment (ArturSharapov) [inline package.json:19]: What about migrations?

File: src/db/supabase/types.ts (hunk 1 above)
Comment (glebchanskiy) [reply src/db/supabase/types.ts:4]: yes, just generated supabase types

File: package.json (hunk 2 above)
Comment (glebchanskiy) [reply package.json:19]: sure, will do it

File: supabase/migrations/20240612140832_data.sql
Comment (glebchanskiy) [reply supabase/migrations/20240612140832_data.sql:1]: I agree, it will need to be corrected

File: .env.local
Comment (ArturSharapov) [inline .env.local:1]: There is [convention](https://vitejs.dev/guide/env-and-mode#env-files) to name files with `.local` if they should be local-only, e.g. because of containing sensitive variables.
```
.env                # loaded in all cases
.env.local          # loaded in all cases, ignored by git
.env.[mode]         # only loaded in specified mode
.env.[mode].local   # only loaded in specified mode, ignored by git
```
> `.env.*.local` files are local-only and can contain sensitive variables. You should add `*.local` to your `.gitignore` to avoid them being checked into git.

Comment (ArturSharapov) [reply .env.local:1]: Since

> An env file for a specific mode (e.g. `.env.production`) will take higher priority than a generic one (e.g. `.env`).

let's just use `.env` for development. We can omit `dotenv -e .env.development --` in this case.

File: .gitignore [hunk 3]
```diff
@@ -16,6 +16,7 @@ pnpm-debug.log*
 # environment variables
 .env
 .env.*
+!.env.local
```
Comment (ArturSharapov) [inline .gitignore:19]:
```suggestion
*.local
```

File: .env.local
Comment (glebchanskiy) [reply .env.local:1]: It was interesting to find out) I'll fix it

File: .env [hunk 4]
```diff
@@ -0,0 +1,11 @@
+VW_ENV=dev
+DATABASE_URL="postgresql://postgres:postgres@127.0.0.1:54322/postgres"
+
+PUBLIC_SUPABASE_URL=http://127.0.0.1:54321
+PUBLIC_SUPABASE_ANON_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZS1kZW1vIiwicm9sZSI6ImFub24iLCJleHAiOjE5ODM4MTI5OTZ9.CRXP1A7WOeoJeXxjNni43kdQwgnWNReilDMblYTn_I0
+SUPABASE_SERVICE_ROLE_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZS1kZW1vIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImV4cCI6MTk4MzgxMjk5Nn0.EGIM96RAZx35lJzdJsyH-qQwv8Hdp7fsn3W0YpN81IU
+JWT_SECRET=super-secret-jwt-token-with-at-least-32-characters-long
```
Comment (ArturSharapov) [inline .env:7]: Loving this secret ❤️

File: package.json [hunk 5]
```diff
@@ -3,20 +3,21 @@
   "type": "module",
   "version": "0.0.1",
   "scripts": {
-    "dev": "dotenv -e .env.development -- astro dev",
-    "start": "dotenv -e .env.development -- astro dev --host",
+    "dev": "dotenv -e .env -- astro dev",
+    "start": "dotenv -e .env -- astro dev --host",
```
Comment (ArturSharapov) [inline package.json:7]: Any reason to keep `dotenv`?

File: src/pages/callback.astro [hunk 6]
```diff
@@ -44,15 +43,30 @@ if (!exists) {
...
+        email: `${userInfo.id}@variants.world`,
+        password: signature,
+        email_confirm: true
+      })
+
+    if (userInSupabaseAuth.user) {
+      await prisma.user.update({ data: { ...userInfo, uuid: userInSupabaseAuth.user.id }, where: { id } })
+    }
+    console.log('User founded, but created in supa!')
+  }
```
Comment (ArturSharapov) [inline src/pages/callback.astro:68]: Move to another PR?

File: supabase/migrations/20240612140832_data.sql
Comment (ArturSharapov) [reply supabase/migrations/20240612140832_data.sql:1]: Shall I set up the workflow in a follow-up branch?

Comment (ArturSharapov) [reply supabase/migrations/20240612140832_data.sql:1]: This file should be deleted either way here.

File: package.json (hunk 5 above)
Comment (glebchanskiy) [reply package.json:7]: yes... otherwise, problems with the prisma DATABASE_URL env

File: supabase/migrations/20240612140832_data.sql
Comment (glebchanskiy) [reply supabase/migrations/20240612140832_data.sql:1]: I didn't quite understand, but this is a backup of 20 posts, with all dependent entities. But it's still a big 1000-line script.

File: src/pages/callback.astro (hunk 6 above)
Comment (glebchanskiy) [reply src/pages/callback.astro:68]: yep

**Review by ArturSharapov — CHANGES_REQUESTED** [review]
- Move login fix to another PR (e.g. to a follow-up one)

File: supabase/migrations/20240612140832_data.sql
Comment (ArturSharapov) [reply supabase/migrations/20240612140832_data.sql:1]: Didn't realize you've trimmed the file already. We can set up the workflow & pulling data from artifacts for dev testing later then.

File: package.json (hunk 5 above)
Comment (ArturSharapov) [reply package.json:7]: Prisma should be working with `.env` file well. The issues you've mentioned are likely related to `.env.development` and `.env.other.stuff`. [Might help](https://stackoverflow.com/questions/67796217/prisma-getting-environment-variable-not-found-error-message-when-running-graph).

**Review by ArturSharapov — APPROVED** [review]
